### ISSUE TITLE
Vulkan rendering frontend

### ISSUE THREAD

**Issue by christianparpart** [contour-terminal/contour#312]
`Vulkan-hpp` denke like an amazing c++ API binding and is even official .

This ticket serves as tracker.

**Comment by WSLUser** [comment]
SDKs are only used for compiling. You use what you need out of it. For optimizing compilations, you'd need to ensure you only build what you need out of the SDK and exclude everything else.

**Comment by christianparpart** [comment]
As much as I'd love to have a vulkan renderer in the frontend, I think I should keep focusing on getting the codebase more robust for the current weeks and some features more usable/stable. :)

**Comment by WSLUser** [comment]
Yeah, I just was answering your question about the SDK. You still need to optimize DirectX before even considering another renderer (unless somebody else takes a stab at it.)

**Comment by christianparpart** [comment]
> Yeah, I just was answering your question about the SDK. You still need to optimize DirectX before even considering another renderer (unless somebody else takes a stab at it.)

i don't have a directx renderer yet BUt a DirectDraw backend for the text shaper / font rasterizer; that one needs to finish in order to bring windows port on par with linux/osx. but thinking about it, I believe Vulkan is also possible on Windows. ;)

**Comment by WSLUser** [comment]
It's possible for all platforms. But since you've already started down the path for DirectX for Windows, I figured you'd want to finish that first.

**Comment by WSLUser** [comment]
So I've discovered Zink, which should help considerably with this and is available in the Mesa package for Linux. Not sure which release got it but 21.x+ at least has it. It should make life much better as it means that Vulkan can be the new rendering front-end but don't have to miss out on features of OpenGL that simply don't exist in other rendering front-ends. I think they have a few Windows issues to deal with still but nothing truly breaking. The point of the project was to increase maintainability and allow developers/projects to more easily adopt Vulkan (such as this one) while keeping some compatibility with older stuff where Vulkan isn't necessarily available due to old hardware or other various factors.

**Comment by data-man** [comment]
I think SDL2 is a very good replacement for Qt.

**Comment by christianparpart** [comment]
> I think SDL2 is a very good replacement for Qt.

What is the benefit /downsides of SDL over Qt from this project's point of view?

**Comment by data-man** [comment]
<table>
<tr>
	<td>
	<td>SDL2
	<td>Qt
<tr>
	<td>License
	<td>ZLIB
	<td>(L)GPL
<tr>
	<td>Easy static linking
	<td>:heavy_check_mark: 
	<td>:x:
<tr>
	<td>Light
	<td>:heavy_check_mark:
	<td>:x:
<tr>
	<td>GLFW
	<td>:x:
	<td>:x:
<tr>
	<td>Surprises
	<td>:x:
	<td>:heavy_check_mark:
</table>

*The table can be updated in the future. :smile:*

**Comment by data-man** [comment]
https://github.com/albin-johansson/centurion - the best C++ wrapper I knew for SDL2 (C++17/20, header-only).
